### PR TITLE
Feature: APP-1752 Multisig DAO - Voting Terminal breakdown and voters

### PR DESCRIPTION
**Pull request by Fabricevladimir**
## Description
This PR
- Adds the voters and breakdown tabs on the Voting Terminal

Task: [APP-1752](https://aragonassociation.atlassian.net/browse/APP-1752)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran all tests with success and extended them if necessary.
- [ ] I have updated the ``CHANGELOG.md`` file in the root folder.
- [x] I have tested my code on the test network.

[APP-1752]: https://aragonassociation.atlassian.net/browse/APP-1752?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by sepehr2github — COMMENTED** [review]
Don't know why, But the app crashed for me in the second page of proposal creation

**Comment by jamesej** [comment]
> Don't know why, But the app crashed for me in the second page of proposal creation

Its a bug that's fixed in latest `develop`. Probably you need to rebase, @Fabricevladimir

File: packages/web-app/src/utils/proposals.ts [hunk 1]
```diff
@@ -38,10 +39,18 @@ import {
...
   SupportedProposals,
 } from './types';
 
-export const MappedVotes: {[key in VoteValues]: VoterType['option']} = {
+export type TokenVotingOptions = StrictlyExclude<
+  VoterType['option'],
+  'approved' | 'none'
+>;
+
+export const MappedVotes: {
```
Comment (WissenIstNacht) [inline packages/web-app/src/utils/proposals.ts:51]: Might have been easier to jus export two sub-types from the UI components :p

File: packages/web-app/src/utils/proposals.ts [hunk 2]
```diff
@@ -521,8 +531,32 @@ export function getTerminalProps(
       )}  ${getFormattedUtcOffset()}`,
     };
   }
+  // This method's return needs to be typed properly
+  else if (isMultisigProposal(proposal)) {
+    voters = [
+      ...new Map(
```
Comment (WissenIstNacht) [inline packages/web-app/src/utils/proposals.ts]: So you go from an array to a Map and then back to an array...?

File: packages/web-app/src/pages/proposal.tsx [hunk 3]
```diff
@@ -332,24 +350,31 @@ const Proposal: React.FC = () => {
 
   // whether current user has voted
   const voted = useMemo(() => {
-    // TODO: updated with multisig
-    if (isMultisigProposal(proposal)) return false;
+    if (!address || !proposal) return false;
```
Comment (WissenIstNacht) [inline packages/web-app/src/pages/proposal.tsx:353]: Do these changes already include the ones Sepehr did a couple of days ago?

File: packages/web-app/src/pages/proposal.tsx [hunk 4]
```diff
@@ -332,24 +350,31 @@ const Proposal: React.FC = () => {
...
+      return proposal.approvals.some(
+        a => a.toLowerCase() === address.toLowerCase()
+      );
+    } else {
+      return proposal.votes.some(
         voter =>
           voter.address.toLowerCase() === address.toLowerCase() &&
           voter.vote !== undefined
-      )
-      ? true
```
Comment (WissenIstNacht) [inline packages/web-app/src/pages/proposal.tsx:344]: Yup thanks, that one bothered me when I last scrolled through this file :D

File: packages/web-app/src/hooks/useWalletCanVote.tsx [hunk 5]
```diff
@@ -34,12 +37,21 @@ export const useWalletCanVote = (
...
+        if (isMultisigClient) {
+          canVote = await (client as MultisigClient)?.methods.canApprove({
+            proposalId: BigInt(stripPlgnAdrFromProposalId(proposalId)),
+            pluginAddress,
+            addressOrEns: address,
+          });
+        } else if (isTokenVotingClient) {
+          canVote = await (client as TokenVotingClient)?.methods.canVote({
+            address,
+            proposalId: stripPlgnAdrFromProposalId(proposalId),
```
Comment (WissenIstNacht) [inline packages/web-app/src/hooks/useWalletCanVote.tsx:51]: No need to BigInt here?

**Comment by WissenIstNacht** [comment]
Thanks a lot!

**Review by WissenIstNacht — COMMENTED** [review]
Thanks a lot!

File: packages/web-app/src/utils/proposals.ts (hunk 2 above)
Comment (Fabricevladimir) [reply packages/web-app/src/utils/proposals.ts]: Yes, array mapped and concatenated two differently typed arrays (voters & all members) into one, converted the resulting array into a Map to remove duplicates, and finally converted the Map back to an array which is what I needed in the first place.

File: packages/web-app/src/pages/proposal.tsx (hunk 3 above)
Comment (Fabricevladimir) [reply packages/web-app/src/pages/proposal.tsx:353]: I rebased so I assume so. Can you confirm @sepehr2github ?

File: packages/web-app/src/hooks/useWalletCanVote.tsx (hunk 5 above)
Comment (Fabricevladimir) [reply packages/web-app/src/hooks/useWalletCanVote.tsx:51]: Nope. This is one area in which I expect changes when we get the new version of the SDK. The proposal ids returned and taken in are wildly inconsistent between clients. For whatever reason, the Multisig will randomly require/return a `BigInt`

File: packages/web-app/src/utils/proposals.ts (hunk 1 above)
Comment (Fabricevladimir) [reply packages/web-app/src/utils/proposals.ts:51]: Ahh, true. Never crossed my mind. That would save having to define the `StrictlyExclude` type

**Review by WissenIstNacht — APPROVED** [review]
Good to go 👍🏼

**Review by sepehr2github — APPROVED** [review]
thanks

**Pull request merged**